clustered/common/src/main/java/org/ehcache/clustered/common/internal/store/operations/TimestampOperation.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.clustered.common.internal.store.operations;
+
+import org.ehcache.clustered.common.internal.store.operations.codecs.CodecException;
+import org.ehcache.spi.serialization.Serializer;
+
+import java.nio.ByteBuffer;
+
+public class TimestampOperation<K, V> implements Operation<K, V> {
+
+  private final K key;
+  private final long timeStamp;
+
+  public TimestampOperation(final K key, final long timeStamp) {
+    if(key == null) {
+      throw new NullPointerException("Key can not be null");
+    }
+    this.key = key;
+    this.timeStamp = timeStamp;
+  }
+
+  TimestampOperation(final ByteBuffer buffer, final Serializer<K> keySerializer) {
+    OperationCode opCode = OperationCode.valueOf(buffer.get());
+    if (opCode != getOpCode()) {
+      throw new IllegalArgumentException("Invalid operation: " + opCode);
+    }
+    this.timeStamp = buffer.getLong();
+    ByteBuffer keyBlob = buffer.slice();
+    try {
+      this.key = keySerializer.read(keyBlob);
+    } catch (ClassNotFoundException e) {
+      throw new CodecException(e);
+    }
+  }
+
+  public K getKey() {
+    return key;
+  }
+
+  @Override
+  public OperationCode getOpCode() {
+    return OperationCode.TIMESTAMP;
+  }
+
+  /**
+   * Remove operation applied on top of another operation does not care
+   * what the other operation is. The result is always gonna be null.
+   */
```

Comment:
That comment doesn't match the implementation.

Reply:
Fixed.